### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/averageAges.js [hunk 1]
```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  // here we filter males for specific century, either we just return males.
+  const male = people.filter(person => !century
```
Comment (masha-girya) [inline src/averageAges.js:19]: maybe it is easier to read when condition parameter specifies like truly from start (i mean person => century)

File: src/averageAges.js [hunk 2]
```diff
@@ -51,11 +63,33 @@ function calculateWomenAverageAge(people, withChildren) {
...
+    // here we find age difference between mothers & childrens
+    return x + child.born - mother.born;
+  }, 0);
+
+  return ageDiff / children.length;
+}
+
+function findAvg(people) {
+  // this is our function for finding an average of people age.
+  const avgAge = people.reduce((x, year) => x + (year.died - year.born), 0);
```
Comment (masha-girya) [inline src/averageAges.js:90]: here i think 'sumOfAge' will be better because you count total age here

Comment (Andriy-Shymkiv) [reply src/averageAges.js:90]: alright

**Review by masha-girya — CHANGES_REQUESTED** [review]
looks nice, its cool that you use function to count average age of people) my remarks are not critical, so you can change it fast and I will aprove)))

**Comment by Andriy-Shymkiv** [comment]
> looks nice, its cool that you use function to count average age of people) my remarks are not critical, so you can change it fast and I will aprove)))

okay

**Review by masha-girya — APPROVED** [review]

File: src/averageAges.js [hunk 3]
```diff
@@ -15,11 +15,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  // here we filter males for specific century, either we just return males.
```
Comment (mykola-koval) [inline src/averageAges.js:18]: remove comments

File: src/averageAges.js [hunk 4]
```diff
@@ -51,11 +63,34 @@ function calculateWomenAverageAge(people, withChildren) {
...
+    const mother = people.find(person => person.name === child.mother);
+
+    // here we find age difference between mothers & childrens
+    return x + child.born - mother.born;
+  }, 0);
+
+  return ageDiff / children.length;
+}
+
+function sumOfAge(people) {
```
Comment (mykola-koval) [inline src/averageAges.js:89]: checklist 1

**Review by mkyloa — CHANGES_REQUESTED** [review]

**Review by illiarerun — APPROVED** [review]

**Pull request closed without merging**